sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -90,6 +90,7 @@ class CSVOptions(
   val quote = getChar("quote", '\"')
   val escape = getChar("escape", '\\')
   val comment = getChar("comment", '\u0000')
+  val lineSeparator = parameters.getOrElse("lineSeparator", "\n")
```

Comment:
Yea, that's why I haven't proposed this change so far (for both hard corded `\n` and configurable line separator).

For the hard corded `\n`, IMHO, reviving hard corded `\n` is probably fine for now as consistent with JSON/TEXT datasources because It uses, after the change I described above, OS-dependent newlline by default in Univocity - https://github.com/uniVocity/univocity-parsers/blob/e34a149077147de7cafcafbbaf8a4310e6297584/src/main/java/com/univocity/parsers/common/Format.java#L78, which is machine-dependent. If we are worried of changing this, we could use `System.getProperty("line.separator")` or just not set via `Option`. This probably does not block this PR.

For the configurable line separator, we are not able to read it back when `wholeFile` (or `miltiLine` now) option is disabled (it's default and this should be primary) because we are dependent of `LineRecordReader` which only deals with `\n` or `\r\n`. We really should be able to read it back.